packages/react-core/src/components/Modal/ModalContent.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ModalBoxBody from './ModalBoxBody';
+import ModalBoxHeader from './ModalBoxHeader';
+import ModalBoxHCloseButton from './ModalBoxCloseButton';
+import ModalBox from './ModalBox';
+import ModalBoxFooter from './ModalBoxFooter';
+import Backdrop from './Backdrop';
+import Bullseye from '../../layouts/Bullseye/Bullseye';
+
+const propTypes = {
+  /** content rendered inside the Modal. */
+  children: PropTypes.node.isRequired,
+  /** additional classes added to the button */
+  className: PropTypes.string,
+  /** Flag to show the modal */
+  show: PropTypes.bool,
+  /** Content of the Modal Header */
+  header: PropTypes.any,
+  /** Content of the Modal Footer */
+  footer: PropTypes.any,
+  /** A callback for when the close button is clicked */
+  onClose: PropTypes.func,
+  /** Creates a large version of the Modal */
+  isLarge: PropTypes.bool,
+  /* Aria label used for Modal Box Header */
+  label: PropTypes.string.isRequired
+};
+
+const defaultProps = {
+  className: '',
+  show: false,
+  header: null,
+  footer: null,
+  onClose: () => undefined,
+  isLarge: false
+};
+
+const ModalContent = ({
+  children,
+  className,
+  show,
+  header,
+  footer,
+  onClose,
+  isLarge,
+  label,
+  ...props
+}) => {
+  const modalBoxHeader = header ? (
+    <ModalBoxHeader> {header} </ModalBoxHeader>
+  ) : null;
+  const modalBoxFooter = footer ? (
+    <ModalBoxFooter> {footer} </ModalBoxFooter>
+  ) : null;
+  if (show) {
+    return (
+      <div {...props}>
+        <Backdrop>
```

Comment:
@jgiardino do we need to close on backdrop click too? I have also seen `esc` key as well.

Reply:
Yes to closing when pressing Esc key.

@mcarrano would have to comment about what the expected behavior is when clicking outside of the modal, and if that also closes it.

Reply:
@jgiardino my guess is that `backdrop` should have option `isStatic` and when set to true clicking outside modal would not close it. At least that's how bootstrap and material is doing it.

Reply:
Agree with @karelhala . Typically I would not expect a modal to close when you click outside of it (on the backdrop). This differentiates it from a popover.

Reply:
So no close on Escape correct?